python/cudf/cudf/core/frame.py

```diff
@@ -2158,7 +2159,9 @@ def _copy_categories(self, other, include_index=True):
                     (cudf.core.index.CategoricalIndex, cudf.MultiIndex),
                 )
             ):
-                self._index._copy_categories(other._index, include_index=False)
+                self._index._postprocess_columns(
+                    other._index, include_index=False
+                )
```

Comment:
Is this a bad find and replace? I'm guessing this should just recurse on `_copy_categories`?

Reply:
Sure was-  fixed